src/utils/code-import.ts

```diff
@@ -0,0 +1,145 @@
+import fs from "node:fs";
+import path from "node:path";
+import { EOL } from "node:os";
+import { visit } from "unist-util-visit";
+import stripIndent from "strip-indent";
+import type { Root, Code, Parent } from "mdast";
+import type { VFile } from "vfile";
+
+interface CodeImportOptions {
+  async?: boolean;
+  preserveTrailingNewline?: boolean;
+  removeRedundantIndentations?: boolean;
+  rootDir?: string;
+  allowImportingFromOutside?: boolean;
+}
+
+function extractLines(
+  content: string,
+  fromLine: number | undefined,
+  hasDash: boolean,
+  toLine: number | undefined,
+  preserveTrailingNewline: boolean = false,
+) {
```

Comment:
Add a return type.

Reply:
the return type is auto inferred, but sure lets add it
```suggestion
): string {
```